packages/flutter/lib/src/rendering/view.dart

```diff
@@ -90,8 +99,12 @@ class RenderView extends RenderObject with RenderObjectWithChildMixin<RenderBox>
     assert(value != null);
     if (configuration == value)
       return;
+    final ViewConfiguration oldConfiguration = _configuration;
     _configuration = value;
-    replaceRootLayer(_updateMatricesAndCreateNewRootLayer());
+    _updateMatrices();
```

Comment:
Why not leave this functionality in _createNewRootLayer? The matrix stored in `_rootTransform` and the one used by the root layer should always be kept in sync, no? And breaking this into two parts makes that invariant harder to keep.

Reply:
Done